tests/rbd/rbd_clone_delete_parent_image.py

```diff
@@ -0,0 +1,129 @@
+from ceph.parallel import parallel
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def rbd_clone_delete_parent_image(rbd, pool_type, **kw):
+    """
+    verify for parent image deletion after flattening the clone and removng snap
+    Args:
+        rbd: rbd object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        **kw: Test data
+    """
+    pool = kw["config"][pool_type]["pool"]
+    image = kw["config"][pool_type]["image"]
+    snap_1 = kw["config"][pool_type].get("snap", f"{image}_snap")
+    clone_1 = kw["config"][pool_type].get("clone", f"{image}_clone")
+    try:
+        # Create multilevel snap and clone
+        snap_created = []
+        clone_created = []
+        rbd.snap_create(pool, image, snap_1)
+        snap_name_1 = f"{pool}/{image}@{snap_1}"
+        snap_created.append(snap_1)
+        rbd.protect_snapshot(snap_name_1)
+        rbd.create_clone(snap_name_1, pool, clone_1)
+        clone_created.append(clone_1)
+
+        # create clone from cloned image
+        snap_2 = clone_1 + "_snap2"
+        clone_2 = clone_1 + "_clone2"
+        rbd.snap_create(pool, image, snap_2)
+        snap_name_2 = f"{pool}/{image}@{snap_2}"
+        snap_created.append(snap_2)
+        rbd.protect_snapshot(snap_name_2)
+        rbd.create_clone(snap_name_2, pool, clone_2)
+        clone_created.append(clone_2)
+
+        # move parent image to trash and flatten the clone and verify the clone flattening
+        rbd.move_image_trash(pool, image)
+        image_id = rbd.get_image_id(pool, image)
+        log.info(f"image id is {image_id}")
+        for clone in clone_created:
+            rbd.flatten_clone(pool, clone)
+
+        # unprotect and delete the snap and verify snap deletion
+        for snap in snap_created:
+            rbd.unprotect_snapshot(snap, pool=pool, image_id=image_id)
+            rbd.snap_remove(pool, image, snap, image_id=image_id)
+
+        # Restore image from trash
+        rbd.trash_restore(pool, image_id)
+
+        # while Running IOs on the image try to delete the image
+        with parallel() as p:
+            p.spawn(
+                rbd.exec_cmd,
+                cmd="rbd bench --io-type write --io-threads 16 --io-total {} {}/{}".format(
+                    "1G", pool, image
+                ),
+            )
+            p.spawn(
```

Comment:
define a method to validate the `rbd rm` command status and error log and spawn it.

Reply:
Added a log message in remove_image method itself.